dashboard/lib/middlewares/redis_session_store.rb

```diff
@@ -18,8 +18,29 @@ def initialize(...)
       super
     end
 
-    # Overrides +Rack::Session::Abstract::Persisted#commit_session?+ to skip committing
-    # unchanged sessions for AJAX requests to reduce Redis traffic.
+    # Overrides +Rack::Session::Redis#delete_session+ to temporarily mark the session as deleted,
+    # allowing concurrent requests to identify that the session is being deleted and prevent its restoration.
+    #
+    # @see https://github.com/redis-store/redis-rack/blob/v3.0.0/lib/rack/session/redis.rb#L54
+    # @see http://web.archive.org/web/20201023020301/https://paulbutcher.com/2007/05/01/race-conditions-in-rails-sessions-and-how-to-fix-them/
+    def delete_session(req, sid, ...)
```

Comment:
What would you think about extracting both the existing logic and this new logic into their own modules, and then including them in the `RedisSessionStore` class? Something like https://gist.github.com/Hamms/f78ad3ad9e896f2f9a280961db135be5, although I'm sure you can come up with some better names.